src/basic/conf-files.c

```diff
@@ -78,7 +78,7 @@ static int files_add(
                 if ((flags & CONF_FILES_FILTER_MASKED))
                         if (null_or_empty(&st)) {
                                 /* Mark this one as masked */
-                                r = set_put_strdup(masked, de->d_name);
+                                r = set_put_strdup(&masked, de->d_name);
```

Comment:
hmm, can you add an `assert(masked);` here before this line? Otherwise this feels as if we'd leak memory here, if people pass NULL (which is OK to do, as long as CONF_FILES_FILTER_MASKED is not set)